### PR TITLE
Test QM selinux permissions

### PR DESCRIPTION
**Pull request by nsednev**
This test case should test inside QM the command setenforce 0 which must fail.

bash-5.1# setenforce 0
setenforce: security_setenforce() failed: Permission denied

What is Selinux?
A test tool insures that inside the QM the command setenforce 0 must fail.

Why?
QM environment should not allow setenforce to change its state and must always fail.
It should be validated via FFI tests.

How the deny is made?
SELinux internal policies prevent a user inside QM change the SELinux setstaus.

How to test?
```
# podman exec -it qm setenforce 0
setenforce:  security_setenforce() failed:  Permission denied

```
or

```
# podman exec -it qm bash
bash-5.1# setenforce 0
setenforce:  security_setenforce() failed:  Permission denied

```

### CODE REVIEW

File: tests/ffi/selinux/test.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# shellcheck disable=SC1091
+
```
Comment (dracher) [inline tests/ffi/selinux/test.sh:4]: This blank line should be removed

File: tests/ffi/selinux/test.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# shellcheck disable=SC1091
+. ../common/prepare.sh
+
+disk_cleanup
+prepare_test
+reload_config
+
```
Comment (Yarboa) [inline tests/ffi/selinux/test.sh:9]: I think you should call 
run_container_in_qm() it starts the container inside qm, later on  you are using podman exec on that container

lets see the test will fail in gihub actions

File: tests/ffi/selinux/test.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+expected_value="setenforce:  security_setenforce() failed:  Permission denied"
+
+# Get setenforce 0 permission denied from qm.
+setenforce_0=$(podman exec -it qm setenforce 0)
+
+echo "This is what qm returns $setenforce_0"
+
+#Check if setenforce 0 succeeds in QM container and fail the test if it does.
+if      [[ $setenforce_0 != *"$expected_value"* ]];then
+        echo "Attempt to change Selinux enforcement to 0 succeeded inside QM container. Test failed."
```
Comment (pengshanyu) [inline tests/ffi/selinux/test.sh:20]: include a clear PASS/FAIL status for each test, something like:
info_message "FAIL: your_function_name: Attempt to change Selinux enforcement to 0 inside QM container"

File: tests/ffi/selinux/test.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+disk_cleanup
+prepare_test
+reload_config
+
+#Preset with expected value the variable.
+expected_value="setenforce:  security_setenforce() failed:  Permission denied"
+
+# Get setenforce 0 permission denied from qm.
+setenforce_0=$(podman exec -it qm setenforce 0)
```
Comment (pengshanyu) [inline tests/ffi/selinux/test.sh:14]: I noticed this comment https://github.com/containers/qm/issues/368#issuecomment-2063909398
@dougsland @Yarboa Do we need to add setenforce 0 into [ffi-tools](https://gitlab.com/CentOS/automotive/container-images/ffi-tools/) ?

File: tests/ffi/selinux/test.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+echo "This is what qm returns $setenforce_0"
+
+#Check if setenforce 0 succeeds in QM container and fail the test if it does.
+if      [[ $setenforce_0 != *"$expected_value"* ]];then
+        echo "Attempt to change Selinux enforcement to 0 succeeded inside QM container. Test failed."
+        exit 1
+fi
+
+echo "Attempt to change Selinux enforcement to 0 denied successfully inside QM container."
```
Comment (pengshanyu) [inline tests/ffi/selinux/test.sh:24]: include a clear PASS/FAIL status for each test, something like:
info_message "PASS: your_function_name: ......"

File: tests/ffi/selinux/test.sh (hunk 2 above)
Comment (nsednev) [reply tests/ffi/selinux/test.sh:9]: I'm just getting the error message from the qm, I don't need to start any containers inside.
Its a really simple test. 
Just running the `podman exec -it qm setenforce 0` and trying to get back `setenforce:  security_setenforce() failed:  Permission denied`

File: tests/ffi/selinux/test.sh (hunk 3 above)
Comment (nsednev) [reply tests/ffi/selinux/test.sh]: Done.

File: tests/ffi/selinux/test.sh (hunk 4 above)
Comment (nsednev) [reply tests/ffi/selinux/test.sh:14]: My code is really basic checkup of the qm, its not even a tool.
Its not even FFI, its more like qm sanity.

File: tests/ffi/selinux/test.sh (hunk 5 above)
Comment (nsednev) [reply tests/ffi/selinux/test.sh]: Done

File: tests/ffi/selinux/test.sh (hunk 2 above)
Comment (Yarboa) [reply tests/ffi/selinux/test.sh:9]: Ahh got it, thanks

File: tests/ffi/selinux/test.sh (hunk 4 above)
Comment (Yarboa) [reply tests/ffi/selinux/test.sh:14]: no, @nsednev is correct

File: tests/ffi/selinux/test.sh (hunk 2 above)
Comment (pengshanyu) [reply tests/ffi/selinux/test.sh:9]: so I think there is no need to call these functions here, right?

File: tests/ffi/selinux/test.sh [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+fi
+
+
+
+
+
+# This is my script that I use for the different repo
+# I would definitely check how other test.sh look in th qm repo, get some inspiration (for example here)
+
+# Specify files that you need and those files are not in your directory
```
Comment (Yarboa) [inline tests/ffi/selinux/test.sh]: Is it needed?

File: tests/ffi/selinux/test.sh (hunk 2 above)
Comment (nsednev) [reply tests/ffi/selinux/test.sh]: Removed them.

File: tests/ffi/selinux/test.sh (hunk 6 above)
Comment (nsednev) [reply tests/ffi/selinux/test.sh]: I don't have those in latest test.sh

**Review by Yarboa — APPROVED** [review]
LGTM

**Comment by pengshanyu** [comment]
should add polarion-id to main.fmf

**Pull request merged**